MM2SpaceCenter.roboFontExt/lib/MM2SpaceCenter.py

```diff
@@ -15,139 +15,147 @@
-        yPos += (lineHeight * 1.2) 
-
-        
-        topLineFields = {
-            "wordCount": [0+leftMargin,   self.wordCount, 20],
-            #"minLength": [108+leftMargin, self.minLength, 3],
-            #"maxLength": [145+leftMargin, self.maxLength, 10],
-        }
+        yPos = 36
+        wordCountInputWidth = 45
+                
         topLineLabels = {
-            "wcText": [31+leftMargin, 78, 'words', 'left'],
-
-            #"wcText": [31+leftMargin, 78, 'words with', 'left'],
-
-           # "lenTextTwo": [133+leftMargin, 10, u'–', 'center'],
-            #"lenTextThree": [176+leftMargin, -0, 'letters', 'left'],
-        }        
-
-        # for label, values in topLineFields.items():
-        #     setattr(self.w, label, EditText((values[0], 0+yPos, 28, 22), text=values[1], placeholder=str(values[2])))
-
-        self.w.wordCount = EditText( (0+leftMargin, 0+yPos, 28, 22), text=self.wordCount, placeholder=self.wordCount, callback=self.wordCountCallback) 
-        
+            "wcText": [leftMargin+wordCountInputWidth+3, 78, 'words', 'left'],
+            "contextLabel": [second_col_x, 78, 'If no words:', 'left'],
+        }   
 
+        self.w.wordCount = NumberEditText((leftMargin, 0+yPos, wordCountInputWidth, 22), text=self.wordCount, callback=self.wordCountCallback, allowFloat=False, allowNegative=False)
+        # EditText( (leftMargin, 0+yPos, wordCountInputWidth, 22), text=self.wordCount, placeholder=self.wordCount, callback=self.wordCountCallback) 
+        self.w.wordCount.getNSTextField().setFocusRingType_(1)
         for label, values in topLineLabels.items():
-            setattr(self.w, label, TextBox((values[0], 3+yPos, values[1], 22), text=values[2], alignment=values[3]))
+            setattr( self.w, label, TextBox( ( values[0], 3+yPos, values[1], 22), text=values[2], alignment=values[3] ) )
+
 
-        
         yPos += lineHeight * 1.3
         
         self.loadDictionaries()
         
         # language selection
         languageOptions = list(self.languageNames)
+        languageOptions.extend(["Any language"])
         
-        
-        
-        self.w.source = PopUpButton((leftMargin, yPos, 85, 20), [], sizeStyle="small", callback=self.changeSourceCallback)
+        self.w.source = PopUpButton((leftMargin, yPos, columnWidth, 20), [], sizeStyle="small", callback=self.changeSourceCallback)
         self.w.source.setItems(languageOptions)
         
         self.w.source.set(4) #default to English for now
         self.source = None 
         self.source = self.w.source.get() #get value, to use for other functions
-
-
-        yPos += lineHeight * 1.2
+        
+        ##### add drop down to choose context: auto, UC, lc, figs, ??, not functional yet, but made as a placeholder
+        
+        self.contextOptions = ['Auto', 'UC', 'LC', 'Figs', 'Frac']
+        self.w.context = PopUpButton((second_col_x, yPos, -leftMargin, 20), [], sizeStyle="small", callback=self.changeContextCallback)
```

Comment:
Second column x position is defined, and the second drop down menu automatically goes to right margin.